src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c

```diff
@@ -188,33 +188,33 @@ void func_808A3D58(BgMoriHineri* this, GlobalContext* globalCtx) {
         this->dyna.actor.draw = BgMoriHineri_DrawHallAndRoom;
         this->actionFunc = func_808A3E54;
 
-        mainCamChildIdx = globalCtx->cameraPtrs[MAIN_CAM]->childCamIdx;
-        if ((mainCamChildIdx != SUBCAM_FREE) && (globalCtx->cameraPtrs[mainCamChildIdx]->setting == CAM_SET_DEMO1)) {
-            OnePointCutscene_EndCutscene(globalCtx, mainCamChildIdx);
+        mainCamChildId = globalCtx->cameraPtrs[CAM_ID_MAIN]->childCamId;
+        if ((mainCamChildId != CAM_ID_MAIN) && (globalCtx->cameraPtrs[mainCamChildId]->setting == CAM_SET_DEMO1)) {
+            OnePointCutscene_EndCutscene(globalCtx, mainCamChildId);
         }
-        OnePointCutscene_Init(globalCtx, 3260, 40, &this->dyna.actor, MAIN_CAM);
-        sNextCamIdx = OnePointCutscene_Init(globalCtx, 3261, 40, &this->dyna.actor, MAIN_CAM);
+        OnePointCutscene_Init(globalCtx, 3260, 40, &this->dyna.actor, CAM_ID_MAIN);
+        sSubCamId = OnePointCutscene_Init(globalCtx, 3261, 40, &this->dyna.actor, CAM_ID_MAIN);
     }
 }
 
 void func_808A3E54(BgMoriHineri* this, GlobalContext* globalCtx) {
     s8 objBankIndex;
 
-    if (globalCtx->activeCamera == sNextCamIdx) {
-        if (sNextCamIdx != MAIN_CAM) {
+    if (globalCtx->activeCamId == sSubCamId) {
+        if (sSubCamId != CAM_ID_MAIN) {
             objBankIndex = this->dyna.actor.objBankIndex;
             this->dyna.actor.objBankIndex = this->moriHineriObjIdx;
             this->moriHineriObjIdx = objBankIndex;
             this->dyna.actor.params ^= 1;
-            sNextCamIdx = MAIN_CAM;
+            sSubCamId = CAM_ID_MAIN;
```

Comment:
I see this type of change in a bunch of other actors but does it make sense for a sub camera ID to be set or compared to `CAM_ID_MAIN`? I feel like this would indicate they could be any camera and not necessarily a sub camera (at least as far as the actor code is concerned)

This makes me wonder if it's even right to be naming main and sub camera IDs differently in most cases, since the difference is already represented by the ID itself and clearly used as such in a lot of places like here

Reply:
This is an important point and part of the discussion as to why the `SUB_CAM_NONE = 0` define was introduced previously, but also later removed.

Basically, for all the variables that store `subCamId`, when it's set to 0, it is no longer used and becomes inactive. If a camera want's to access the main camera, it will always use `CAM_ID_MAIN` directly instead of going through the `subCamId` temps. So `subCamId` is never actually used for the main cam, it's always used through a temp. It is this idea of inactivity which was the reason `SUB_CAM_NONE` existed in the first place.

Likewise, when a subcamera is requested, it is checked which of the 3 subcamera's are available and assigns that to `subCamId`. It's not always the same subCam that get's assigned, basically it's the first one that's available (that way the game can handle multiple cutscenes at once). It seems distinct enough from the main camera to warrant it's own name to me imo, but there is a bit of a gray area here.

Reply:
Is there a real distinction between main and sub camera ids as far as the camera system is concerned? I feel like these `subCamId` variables could also be called `camId` and it would still make sense (`camId` being the id of the separate camera used in the context of the actor for instance), but I might be missing some internal details.

Otherwise, based on your explanation, I don't think anything treated as a "sub camera id" should be compared or set to `CAM_ID_MAIN`, so I guess we should at least use 0 or a dedicated define like the `SUB_CAM_NONE` you mentioned.

Basically I think we should either treat a camera id as a single thing and have no difference between sub and main camera ids, or completely separate them with different values/defines.

Reply:
Bump on this discussion as this was the main sticking point that I was split 50/50 on, and was a main factor in my delayed response. We can move this discussion to discord and I can copy the message below for reference. But for now:

The main point I was stuck on was whether to consider the 4 cameras as either:
- "4 identical camera" with the main vs. sub distinction only being in the application of the cameras
- "1 main camera + 3 sub cameras" with making a stronger differentiation between these.

Points for "4 identical camera"
- All 4 camera's have the same struct
- Main vs. sub is more of how it's used vs the struct itself

Points for "1 main camera + 3 sub camera"
- The "main camera" is clearly designed for normal game-play as link, while the 3 sub-cameras are clearly designed for cutscenes of various sorts, you can look at all the actors from this PR to see cases of this
- There are cases, particularly in "z_play", that treat the main cam different from the 3 sub-cams, particularly: https://github.com/zeldaret/oot/blob/32e66c2da8c2f9816773e1b2a0bc92804e9d20de/src/code/z_play.c#L218-L225 and the function `Gameplay_CreateSubCamera` https://github.com/zeldaret/oot/blob/32e66c2da8c2f9816773e1b2a0bc92804e9d20de/src/code/z_play.c#L1505-L1512

Reply:
Going back to this discussion, I think it makes sense to treat main and sub cameras separately in the global context code (ie. the `Gameplay_` camera functions), because they clearly separated them on a lower level like you mentioned.

But when it comes to the rest of the codebase, I think it *could* make more sense to omit the distinction in the names, and rely on the camera ID to make the distinction between main camera (ID 0) and sub cameras (ID 1+).

However, if we do want to distinguish main and sub cameras everywhere, then I still think we should have a separate `SUB_CAM_ID_NONE` for this purpose (or maybe `SUB_CAM_ID_STOP` or `SUB_CAM_ID_DONE` which would avoid confusion with `CAM_ID_NONE`). Setting a sub cam id to main "works" if you already know what it means, but it still feels very wrong to me and would be very confusing for someone not familiar with the pattern.

I'm personally okay with either options, and I'm not sure which one would be better since both standpoints can be argued, but I really think we need to make a change in some direction